Employees/PermanentEmployeeJSON.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Employees.Entites;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Employees
+{
+    public class PermanentEmployeeJSON
+    {
+        public string path = @"..\..\..\..\Employees\Data\PermanentEmployee.json";
+        public PermanentEmployee PermanentEmployee5 = new PermanentEmployee("Alice surname", 50000, 5000, holidayAllowance: 19);
+        public PermanentEmployee PermanentEmployee6 = new PermanentEmployee("Jess", 2700, 400, 17);
+
+        public void WriteToEmployeeFile(PermanentEmployee pe)
+        {
+            if (!File.Exists(path))
+            {
+                File.Create(@"..\..\..\..\Employees\Data\ContingencyPermanentEmployee.json");
```

Comment:
This path works because you're running your program within the context of your repository, so it goes up the tree outside of the program folder and down into your class library project. 

What if your program were installed on some other computer? The class library project wouldn't be there, and as a dev coming to debug it you might not realise the "Employees" folder next to the app was actually part of it. 

It would be better to either:
- use a subfolder under the folder where the application is running, or 
- use a configuration setting to set the folder, so that it can be different on that other computer and so that the developer coming to it has a trail to follow in the config file to help them find that folder

Reply:
Do you mind elaborating on these solutions?

Reply:
Sure. 

You can find out where your application is running without hard-coding a path, and then use a file within that folder, by doing something like this:

```csharp
using System;
using System.IO;

var whereIsThisRunning = AppDomain.CurrentDomain.BaseDirectory;
var relativePathFromThereToTheData = "..\data\some.json";
var pathToDataFile = Path.GetFullPath(Path.Combine(whereIsThisRunning, relativePathFromThereToTheData));
```

On the second suggestion, you can have an `appsettings.json` file in your project with settings in that where you can change them without recompiling the code. That makes it easy to change when you run the program somewhere else, such as in production. When your program runs it reads the settings from the file each time it needs them. 

See this page on [Configuration in .NET](https://docs.microsoft.com/en-us/dotnet/core/extensions/configuration) for some standard code you can add that lets your program read from `appsettings.json`. You don't need to write your own JSON reading code for appsettings.json as you've done before, because it's built into .NET. That page has also got a lot more detail about how else you can store configuration information if you want to dive deeper, but get the simple example working first or the rest of it could get confusing.